### PR TITLE
VH 1303 - Updated the script to generate a .env file based on user input

### PR DESCRIPTION
**Pull request by DokurOmkar**
# PR Details
## Description

In this PR, initialization script is updated to generate a .env file based on user input. 

## Related Issue
[VH-1303](https://usdot-carma.atlassian.net/browse/VH-1303)

## How Has This Been Tested?
This update has been tested by running the script locally to check that user input is taken and .env file created.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[V2XHUB Contributing Guide](https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: configuration/initialization.sh [hunk 1]
```diff
@@ -1,4 +1,82 @@
 #!/bin/bash
+
+# Default values for the variables
+PORT_DRAYAGE_ENABLED_DEFAULT="TRUE"
```
Comment (paulbourelly999) [inline configuration/initialization.sh:4]: Change this to false by default

File: configuration/initialization.sh [hunk 2]
```diff
@@ -9,15 +87,15 @@ sudo apt update -y && sudo apt upgrade -y
 sudo apt-get install chromium-browser -y
 
 # Make passwords for mysql
-mkdir -p secrets && cd secrets
+mkdir -p secrets && cd secrets || return
```
Comment (paulbourelly999) [inline configuration/initialization.sh:90]: Why is this necessary. Is this something from shell check. Can you add justification for adding return statement here, maybe even link rule.

File: configuration/initialization.sh [hunk 3]
```diff
@@ -26,15 +104,14 @@ fi
 if test -f "$FILE2"; then
     echo "$FILE2 exists."
 else
-    read -p "enter password for mysql_password: " sql_pass
+    read -r -p "enter password for mysql_password: " sql_pass
     echo "$sql_pass" > sql_pass.txt
     # Remove endline characters from password files
     tr -d '\n' <sql_pass.txt> mysql_password.txt && rm sql_pass.txt
 fi
-# TODO VH-1303 Allow for version and configuration selection in initialization script
```
Comment (paulbourelly999) [inline configuration/initialization.sh:34]: Great catch!

File: configuration/initialization.sh [hunk 4]
```diff
@@ -50,10 +127,19 @@ echo \
...
+elif [[ $PORT_DRAYAGE_ENABLED == "FALSE" ]]; then
+    sudo docker compose up php -d
+else
+    echo "Invalid value for PORT_DRAYAGE_ENABLED. Please provide either TRUE or FALSE."
+    exit 1
+fi
 
 # Create V2X Hub user
-cd $mysqlDir
+cd "$mysqlDir" || return
```
Comment (paulbourelly999) [inline configuration/initialization.sh:142]: Why is this necessary?

File: configuration/initialization.sh [hunk 5]
```diff
@@ -1,4 +1,82 @@
...
+    # Simulation IP
+    read -r -p "Enter Simulation IP (or press Enter to use default is $SIMULATION_IP_DEFAULT): " SIMULATION_IP
+    SIMULATION_IP=${SIMULATION_IP:-$SIMULATION_IP_DEFAULT}
+
+    # Sensor Configuration File Path
+    read -r -p "Enter Sensor Configuration File Path (or press Enter to use default as $SENSOR_JSON_FILE_PATH_DEFAULT): " SENSOR_JSON_FILE_PATH
+    SENSOR_JSON_FILE_PATH=${SENSOR_JSON_FILE_PATH:-$SENSOR_JSON_FILE_PATH_DEFAULT}
+fi
+
+# Write to .env file
```
Comment (paulbourelly999) [inline configuration/initialization.sh:65]: If .env file already exists we may want to warn user that we are overwriting of even ask for permission to overwrite

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
Some small comments but otherwise this looks great! Thanks

File: configuration/initialization.sh [hunk 6]
```diff
@@ -1,4 +1,82 @@
 #!/bin/bash
+
+# Default values for the variables
+PORT_DRAYAGE_ENABLED_DEFAULT="TRUE"
+INFRASTRUCTURE_ID_DEFAULT="rsu_1234"
+INFRASTRUCTURE_NAME_DEFAULT="East Intersection"
+V2XHUB_IP_DEFAULT="127.0.0.1"
+SIMULATION_MODE_DEFAULT="FALSE"
+SIMULATION_IP_DEFAULT="127.0.0.1"
```
[the remark this answers is not in the document]
Comment (DokurOmkar) [reply configuration/initialization.sh:9]: Yes, we just define a default that could be used if the user enables the simulation mode on line 55:

if [[ $SIMULATION_MODE == "TRUE" ]]; then

File: configuration/initialization.sh (hunk 1 above)
Comment (DokurOmkar) [reply configuration/initialization.sh:4]: Changed it to False

File: configuration/initialization.sh (hunk 2 above)
Comment (DokurOmkar) [reply configuration/initialization.sh:90]: Yes, shell has thrown a warning here "^-- [SC2164](https://www.shellcheck.net/wiki/SC2164) (warning): Use 'cd ... || exit' or 'cd ... || return' in case cd fails."

Added a comment next to return saying "# return in case cd fails"

File: configuration/initialization.sh [hunk 7]
```diff
@@ -9,15 +87,15 @@ sudo apt update -y && sudo apt upgrade -y
...
+mkdir -p secrets && cd secrets || return
 
 # Creates password files where user inputs password
 FILE1=mysql_root_password.txt
 FILE2=mysql_password.txt
 if test -f "$FILE1"; then
     echo "$FILE1 exists."
 else
-    read -p "enter password for the mysql_root_password: " sql_root_pass
+    read -r -p "enter password for the mysql_root_password: " sql_root_pass
```
[the remark this answers is not in the document]
Comment (DokurOmkar) [reply configuration/initialization.sh:98]: Added this again as shell check thrown an error:

^-- [SC2162](https://www.shellcheck.net/wiki/SC2162) (info): read without -r will mangle backslashes.

For this reason it is "read -r -p"

File: configuration/initialization.sh (hunk 4 above)
Comment (DokurOmkar) [reply configuration/initialization.sh:142]: Added a comment next to return saying "# return in case cd fails"

File: configuration/initialization.sh (hunk 5 above)
Comment (DokurOmkar) [reply configuration/initialization.sh]: Updated the script to ask for permission to overwrite the existing .env file

File: configuration/initialization.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (paulbourelly999) [reply configuration/initialization.sh:9]: Sounds good

File: configuration/initialization.sh [hunk 8]
```diff
@@ -9,15 +93,15 @@ sudo apt update -y && sudo apt upgrade -y
 sudo apt-get install chromium-browser -y
 
 # Make passwords for mysql
-mkdir -p secrets && cd secrets
+mkdir -p secrets && cd secrets || return # return in case cd fails
```
Comment (paulbourelly999) [inline configuration/initialization.sh:96]: Can you link the rule

File: configuration/initialization.sh (hunk 7 above)
[the remark this answers is not in the document]
Comment (paulbourelly999) [reply configuration/initialization.sh:104]: Can you comment the rule

**Review by paulbourelly999 — APPROVED** [review]

File: configuration/initialization.sh (hunk 7 above)
[the remark this answers is not in the document]
Comment (DokurOmkar) [reply configuration/initialization.sh]: Added the Rule as a comment

File: configuration/initialization.sh (hunk 8 above)
Comment (DokurOmkar) [reply configuration/initialization.sh]: Added the Rule as a comment

File: configuration/initialization.sh [hunk 9]
```diff
@@ -50,10 +133,19 @@ echo \
   sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
 sudo apt-get update
 sudo apt-get -y install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
-sudo docker compose up -d
+
+# Check if Port Drayage functionality is enabled
+if [[ $PORT_DRAYAGE_ENABLED == "TRUE" ]]; then
+    sudo docker compose up -d
+elif [[ $PORT_DRAYAGE_ENABLED == "FALSE" ]]; then
```
Comment (jwillmartin) [inline configuration/initialization.sh:140]: Most of the time, users or at least I will not want port drayage enabled. Won't this only bring up the php container?
